### PR TITLE
Handle witness signatures

### PR DESCRIPTION
**Pull request by simone-kalbermatter**
This PR adds a handler for witness signature messages. Witness messages are now ordered by newest on top to oldest on the bottom. After signing a witness message, the sign button is disabled and shows the text _signed_. The witness signatures are now visible in the signatures dropdown menu.

I also changed the description of the election setup witness message to display all election questions.

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/WitnessingHandler.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    PublicKey witnessPk = context.getSenderPk();
+
+    // Check that the sender of the message is a witness
+    if (!laoView.getWitnesses().contains(witnessPk)) {
+      throw new InvalidWitnessException(witnessMessageSignature, witnessPk);
+    }
+
+    // Check that a witness message with the given message id exists in the lao
+    MessageID messageID = witnessMessageSignature.getMessageId();
+    WitnessMessage witnessMessage = laoView.getWitnessMessages().get(messageID);
```
Comment (matteosz) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/WitnessingHandler.java:45]: As specified in the issue #1374 the LAO should not handle the witness messages, so a custom repository would be far better, but it's your choice whether to postpone it for a future PR or do it now

**Review by matteosz — CHANGES_REQUESTED** [review]
Very good work! Just few suggestions

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/witness/WitnessMessageListViewAdapter.java [hunk 2]
```diff
@@ -111,8 +112,20 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
+      Set<PublicKey> witnessSignatures = witnessMessage.getWitnesses();
+
+      if (witnessSignatures.contains(laoViewModel.getPublicKey())) {
+        // The user already signed the message
+        binding.signMessageButton.setEnabled(false);
+        binding.signMessageButton.setText("Signed");
+      } else {
+        binding.signMessageButton.setEnabled(true);
+        binding.signMessageButton.setText("Sign");
+        View.OnClickListener listener =
```
Comment (matteosz) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/witness/WitnessMessageListViewAdapter.java:124]: Last thing that I forgot, could you use string resources here? As a general principle each displayed string should be an Android resource

**Review by matteosz — CHANGES_REQUESTED** [review]

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/WitnessingHandler.java (hunk 1 above)
Comment (simone-kalbermatter) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/WitnessingHandler.java:45]: Yes I saw that, but decided to keep the larger refactoring of adding the witness repo for another PR.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Lao.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public Lao(
     this.organizer = organizer;
     this.modificationId = modificationId;
     this.witnesses = new HashSet<>(witnesses);
-    this.witnessMessages = new HashMap<>(witnessMessages);
+    this.witnessMessages = new LinkedHashMap<>(witnessMessages);
```
[the remark this answers is not in the document]
Comment (simone-kalbermatter) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Lao.java:80]: The `LinkedHashMap` works for ordering the messages, the problem was caused by dropping signed messages to the bottom of the list. This caused inconsistencies depending on if the witness messages were signed one after another or in a random order. I changed the order now to be fixed, regardless wether the message was signed or not. The oldest messages are now on the bottom, and newest on top.

**Review by matteosz — APPROVED** [review]
LGTM!

**Review by Warhedgehog — APPROVED** [review]
Great Work, well documented and tested !

**Pull request merged**